aten/src/ATen/native/quantized/cpu/qlinear.cpp

```diff
@@ -140,7 +140,11 @@ class QLinearInt8 final : public torch::OperatorKernel {
         /*thread_id=*/0,
         /*num_threads=*/1);
 
-    return output;
+    // The resulting matrix here is 2-D, let's view it with the original
```

Comment:
Why is this needed? qLinear works for Lenet without this change: can you provide more context?

Reply:
at::linear can take inputs of dimensions >2

it also highlights point about testing - we should see whether we can auto-convert relevant test_nn tests into quantized tests to catch situations like this.

In the meantime - can we please just add a unittest in this PR?

Reply:
- Allocate the tensor of the right shape in the `at::zeros` function, instead of using `view` function.
- Add the comments with two examples to give more context and better elaboration for the cases we want to handle.
- Add the unit test in this PR.